packages/legend-query-builder/src/components/shared/BasicValueSpecificationEditor.tsx

```diff
@@ -601,29 +603,61 @@ const CollectionValueInstanceValueEditor = observer(
 
     if (editable) {
       return (
-        <div className={clsx('value-spec-editor', className)}>
-          <input
-            ref={inputRef}
-            className="panel__content__form__section__input value-spec-editor__input"
-            spellCheck={false}
-            value={text}
-            placeholder={text === '' ? '(empty)' : undefined}
-            onChange={changeValue}
-          />
-          <button
-            className="value-spec-editor__list-editor__save-button btn--dark"
-            onClick={saveEdit}
-          >
-            <SaveIcon />
-          </button>
-          <button
-            className="value-spec-editor__reset-btn"
-            title="Reset"
-            onClick={resetValue}
-          >
-            <RefreshIcon />
-          </button>
-        </div>
+        <>
+          {showAdvancedEditorPopover && (
```

Comment:
Looking at the video preview, I think when we're showing the popover, we should not show anything in the `input` field, it looks odd, you can even consider moving the popover to completely eclipse the original input box, by manipulating `menuProps.anchorOrigin and menuProps.transformOrigin`. Also, add a small box at the bottom of the popover to say - hit `Enter` to apply change, `Esc` and `click-out` action should not save the result <-- this is also the behavior in `vscode`, look at vscode when you try to create a new file in the explorer tree sidebar, that's kind of the visual guide and interaction I hope we have.